pkg/mic/mic.go

```diff
@@ -1317,3 +1327,94 @@ func (c *Client) checkIfIdentityImmutable(id string) bool {
 	}
 	return false
 }
+
+func (c *Client) generateIdentityAssignmentState() (currentState map[string]map[string]bool, desiredState map[string][]string, isVMSSMap map[string]bool, err error) {
+	assignedIDs, err := c.CRDClient.ListAssignedIDs()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to list AzureAssignedIdentities, error: %+v", err)
+	}
+
+	isVMSSMap = make(map[string]bool)
+	currentState = make(map[string]map[string]bool)
+	desiredState = make(map[string][]string)
+	for _, assignedID := range *assignedIDs {
+		node, err := c.NodeClient.Get(assignedID.Spec.NodeName)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to get node %s, error %+v", assignedID.Spec.NodeName, err)
+		}
+
+		nodeName, isVMSS, err := isVMSS(node)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to check if node %s is VMSS, error: %+v", assignedID.Spec.NodeName, err)
+		} else if isVMSS {
+			nodeName = getVMSSName(nodeName)
+		} else {
+			// VM node name does not require conversion
+			nodeName = assignedID.Spec.NodeName
+		}
+		isVMSSMap[nodeName] = isVMSS
+
+		if assignedID.Status.Status == aadpodid.AssignedIDAssigned {
```

Comment:
Lets add a comment here on why we pick only the identities with status assigned and why we ignore the rest